docs/developer-guide/syntax-reference.rst

```diff
@@ -795,6 +800,11 @@ for deletion and if the topic format is AVRO, delete the corresponding Avro sche
 deleted, too. Topic deletion is asynchronous, and actual removal from brokers
 may take some time to complete.
 
+.. note:: DELETE TOPIC will not necessarily work if your kafka cluster is configured
+  to create topics automatically with ``auto.create.topics.enable=true``. We
+  recommended checking after a few minutes to ensure that the topic was
+  deleted.
```

Comment:
How will the user know this did not work? is the `auto.create.topics.enable=true` flaky when deleting the topic? Why a few minutes, btw? Shouldn't this happen immediately?

Reply:
`auto.create.topics.enable` means that the streams task will automatically create the topic when something is published to that topic. this means that the delete will successfully delete the topic, but since we haven't stopped the streams task, if any message is published it will re-create the topic. 

there is a short delay between enqueuing the message on the command topic and terminating the streams task. in most cases, you will terminate the query before you drop it - so this will only happen if there are multiple nodes and one is lagging (hence I suggest waiting a while).

Reply:
Here is the illustration of a scenario it doesn't work:
```sql
1: CREATE STREAM FOO AS SELECT * FROM BAR;
2: Server A starts CSAS_FOO_1
3: Server B starts CSAS_FOO_1
4: TERMINATE CSAS_FOO_1; DROP FOO DELETE TOPIC;
5: Server A terminates CSAS_FOO_1;
6: Server A deletes topic FOO (ON REST SIDE);
7: Server A drops FOO from metastore
8: Server B re-creates topic FOO due to auto.create
9: Server B terminates CSAS_FOO_1
10: SERVER B drops FOO form metastore
```

Reply:
Oh, it is a synchronization problem. Should we have some explanation about it in the syntax-reference? I understand what you say now, but new users or users who have only 1 server might not understand this behavior, right?

Reply:
@JimGalasyn I'd be interested to hear your thoughts. 

I'm happy adding documentation for this, but I'm worried that adding context might just confuse a user who is happy to just follow our guidance (since even describing the problem precisely is difficult). It is not an issue with just multiple servers, it is also possible that within a single server the consumer of the command topic is delayed, in which case the same scenario as above can still happen.

Reply:
@agavra In general, if a feature works, it goes into docs, and if there's breakage, it goes into KB. In this case, we might want to call out this situation in the [Troubleshooting](https://docs.confluent.io/current/ksql/docs/troubleshoot-ksql.html) topic, if we think a significant number of people will run across it.

Reply:
KB sounds sensible, with a link to the item from the docs, ... do we have a KB???? :D

Reply:
I didn't know we had a KB (does that stand for Knowledge Base?) 😉